far/viewer.cpp

```diff
@@ -1007,11 +1011,20 @@ void Viewer::SetStatusMode(int Mode)
 	ShowStatusLine=Mode;
 }
 
+static constexpr std::array extra_word_divs{ encoding::bom_char, encoding::replace_char };
 
 static bool is_word_div(const wchar_t ch)
 {
-	static const wchar_t extra_div[]{ encoding::bom_char, encoding::replace_char };
-	return !ch || std::iswspace(ch) || contains(Global->Opt->strWordDiv.Get(), ch) || contains(extra_div, ch);
+	return !ch || std::iswspace(ch) || contains(Global->Opt->strWordDiv.Get(), ch) || contains(extra_word_divs, ch);
+}
+
+static const string& get_word_div()
+{
+	static string word_div;
```

Comment:
Should it really be static?

We rewrite it on every invocation anyway (which is correct - Opt->strWordDiv might change), so the only benefit is reusing the allocated storage.

Reply:
That was the reason. Why make an allocation when we can avoid it altogether with zero overhead? If this function would be defined in `string_utils` or such, where it could be called in unrelated scenario while the buffer is still in use, that would be a different story. As it is, I think a warning comment here is sufficient. Please let me know whether you prefer returning a newly allocated string every time or just add a comment.

Reply:
I'd rather not premature optimize object lifetimes unless necessary, but if it is called many times it is probably reasonable.

However, should it be called multiple times?
Looks like it is used in search_text_forward & search_text_backward, called by Search in a loop. 
While we should query Opt->strWordDiv before every search to make sure it is relevant, there is probably no need to do so on every iteration within a search.
Perhaps get_word_div() can be a class member `string m_WordDiv`, updated in `Viewer::Search` only?

Reply:
That's a good point. Changed to call it once per search only in Text mode.